PlayList.java

```diff
@@ -71,34 +98,85 @@ public int indexOf(String title) {
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
     public boolean add(int i, Track track) {
-        //// replace the following statement with your code
-        return false;
+        if(this.getSize() < i || i < 0 || this.getSize() == this.maxSize){
+            return false;
+            }
+        else{
+            for(int j = size ; j > i ; j--){
+                tracks[j] = tracks[j-1];
+            }
+            tracks[i] = track;
+            size = size + 1;
+        }
+        return true;
     }
      
     /** Removes the track in the given index from this list.
      *  If the list is empty, or the given index is negative or too big for this list, 
      *  does nothing and returns -1. */
     public void remove(int i) {
-        //// replace this comment with your code
+        if((this.getSize() - 1) < i || i < 0 || this.getSize() == 0){
+            return;
+            }
+        else{
+            tracks[i] = null;
+            for(int j = i ; j < size - 1 ; j++){
+                tracks[j] = tracks[j+1];
+            }
+            size = size - 1;
+
+        }
     }
 
     /** Removes the first track that has the given title from this list.
      *  If such a track is not found, or the list is empty, or the given index
      *  is negative or too big for this list, does nothing. */
     public void remove(String title) {
-        //// replace this comment with your code
+        String subtitle = title.substring(1);
+        subtitle = subtitle.toLowerCase();
+        String first = title.substring(0, 1);
+        first = first.toUpperCase();
+        title = first + subtitle;
+        int index = this.getSize() + 1;
+        if(size == 0){
+            return;
+        }
+        for(int i = 0 ; i < this.getSize() ; i++){
+            if(title.equals(tracks[i].getTitle())){
+              index = i;
+              break;
+            }
+         }
+         if(index > size || size == 0){
+            return;
+         }
```

Comment:
Could be done much more easier by using `indexOf`
```suggestion
        int index = indexOf(title);
```